pkgs/os-specific/linux/kexec-tools/default.nix

```
@@ -18,7 +18,7 @@ stdenv.mkDerivation rec {
       url = "https://raw.githubusercontent.com/void-linux/void-packages/6c1192cbf166698932030c2e3de71db1885a572d/srcpkgs/kexec-tools/patches/ppc64-elfv2.patch";
       sha256 = "19wzfwb0azm932v0vhywv4221818qmlmvdfwpvvpfyw4hjsc2s1l";
     })
-  ];
+  ] ++ lib.optional (targetPlatform.useLLVM or false) ./fix-purgatory-llvm-libunwind.patch;
```

```suggestion
  ] ++ lib.optional (stdenv.buildPlatform.useLLVM or false) ./fix-purgatory-llvm-libunwind.patch;
```

shouldn't this be buildPlatform since the linker is getting run on the buildHost?
```
nix-repl> pkgsLLVM.buildPlatform.useLLVM
false
```
huh -- that is unexpected. i'm confused.

---

Switched to `stdenv.hostPlatform`, should be good enough.

---

idk sill seems weird as hostPlatform is where the code will run
https://nix.dev/tutorials/cross-compilation
in other places we used `stdenv.cc.binutils.isLLVM` to query if the linker is `lld` tho there is also `stdenv.{build,host,target}Platform.linker` which resolves to `lld`. did have an approved upstream PR to fix https://github.com/llvm/llvm-project/pull/94965 but was unable to merge due to an email requirement.

---

>  did have an approved upstream PR to fix [llvm/llvm-project#94965](https://github.com/llvm/llvm-project/pull/94965) but was unable to merge due to an email requirement.

update: maintainer merged a fix https://github.com/llvm/llvm-project/commit/8cc6a2469c95f0815487c442495a829c337913c4 so shouldn't be an issue for llvm19

doesn't effect this change tho.